### PR TITLE
[APPS-2130] moment to date-fns migration

### PR DESCRIPTION
**Pull request by pkunduGL**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

moment dependency is being used in the mentioned components (base-edit-task-filter-cloud.component,edit-service-task-filter-cloud.component and edit-task-filter-cloud.component )

**What is the new behaviour?**
removed the moment js dependcies and updated with date-fns

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://alfresco.atlassian.net/browse/APPS-2130

### CODE REVIEW

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts [hunk 1]
```diff
@@ -285,11 +291,11 @@ export abstract class BaseEditTaskFilterCloudComponent<T> implements OnInit, OnC
...
     onDateChanged(newDateValue: any, dateProperty: TaskFilterProperties) {
         if (newDateValue) {
-            const momentDate = moment(newDateValue, BaseEditTaskFilterCloudComponent.FORMAT_DATE, true);
+            const date = DateFnsUtils.formatDate(newDateValue, BaseEditTaskFilterCloudComponent.FORMAT_DATE);
             const controller = this.getPropertyController(dateProperty);
 
-            if (momentDate.isValid()) {
-                controller.setValue(momentDate.toISOString(true));
+            if (isValid(date)) {
+             controller.setValue( new Date(date).toISOString() );
```
Comment (rbahirsheth) [inline lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts:298]: remove the trailing & initial space inside the bracket.

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/edit-task-filter-cloud.component.ts [hunk 2]
```diff
@@ -88,13 +89,8 @@ export class EditTaskFilterCloudComponent extends BaseEditTaskFilterCloudCompone
...
     private setLastModifiedToFilter(formValues: TaskFilterCloudModel) {
         if (formValues.lastModifiedTo && Date.parse(formValues.lastModifiedTo.toString())) {
-            const lastModifiedToFilterValue = moment(formValues.lastModifiedTo);
-            lastModifiedToFilterValue.set({
-                hour: 23,
-                minute: 59,
-                second: 59
-            });
-            formValues.lastModifiedTo = lastModifiedToFilterValue.toISOString(true);
+            const lastModifiedToFilterValue =  set(new Date(formValues.lastModifiedTo), { hours: 23, minutes: 59,seconds: 59});
```
Comment (rbahirsheth) [inline lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/edit-task-filter-cloud.component.ts:92]: add space before '}'

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts [hunk 3]
```diff
@@ -20,18 +20,20 @@ import { AssignmentType, FilterOptions, TaskFilterAction, TaskFilterProperties,
...
 import { TaskFilterDialogCloudComponent } from '../task-filter-dialog/task-filter-dialog-cloud.component';
 import { MatDialog } from '@angular/material/dialog';
 import { IdentityUserModel } from '../../../../people/models/identity-user.model';
 import { IdentityGroupModel } from '../../../../group/models/identity-group.model';
 import { MatSelectChange } from '@angular/material/select';
 import { Environment } from '../../../../common/interface/environment.interface';
+import {isValid} from 'date-fns';
+import { DateFnsAdapter } from '@angular/material-date-fns-adapter';
+import { DatetimeAdapter } from '@mat-datetimepicker/core';
+
```
[the remark this answers is not in the document]
Comment (pkunduGL) [reply lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts:36]: removed extra line

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts [hunk 4]
```diff
@@ -149,9 +151,13 @@ export abstract class BaseEditTaskFilterCloudComponent<T> implements OnInit, OnC
         this.userPreferencesService
             .select(UserPreferenceValues.Locale)
             .pipe(takeUntil(this.onDestroy$))
-            .subscribe((locale) => this.dateAdapter.setLocale(locale));
+            .subscribe(locale => this.setLocale(locale)
+            );
     }
 
+    setLocale(locale) {
```
[the remark this answers is not in the document]
Comment (pkunduGL) [reply lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts]: updated

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/edit-task-filter-cloud.component.ts (hunk 2 above)
Comment (pkunduGL) [reply lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/edit-task-filter-cloud.component.ts]: added

**Review by rbahirsheth — APPROVED** [review]

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts [hunk 5]
```diff
@@ -20,18 +20,19 @@ import { AssignmentType, FilterOptions, TaskFilterAction, TaskFilterProperties,
...
-import { DateAdapter } from '@angular/material/core';
-import { TranslationService, UserPreferencesService, UserPreferenceValues } from '@alfresco/adf-core';
+import { DateFnsUtils, TranslationService, UserPreferencesService, UserPreferenceValues } from '@alfresco/adf-core';
 import { TaskFilterDialogCloudComponent } from '../task-filter-dialog/task-filter-dialog-cloud.component';
 import { MatDialog } from '@angular/material/dialog';
 import { IdentityUserModel } from '../../../../people/models/identity-user.model';
 import { IdentityGroupModel } from '../../../../group/models/identity-group.model';
 import { MatSelectChange } from '@angular/material/select';
 import { Environment } from '../../../../common/interface/environment.interface';
+import {isValid} from 'date-fns';
```
Comment (rbahirsheth) [inline lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts:33]: give space around isValid.

Comment (pkunduGL) [reply lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts]: Added proper space

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts [hunk 6]
```diff
@@ -20,18 +20,19 @@ import { AssignmentType, FilterOptions, TaskFilterAction, TaskFilterProperties,
...
+import { DateFnsUtils, TranslationService, UserPreferencesService, UserPreferenceValues } from '@alfresco/adf-core';
 import { TaskFilterDialogCloudComponent } from '../task-filter-dialog/task-filter-dialog-cloud.component';
 import { MatDialog } from '@angular/material/dialog';
 import { IdentityUserModel } from '../../../../people/models/identity-user.model';
 import { IdentityGroupModel } from '../../../../group/models/identity-group.model';
 import { MatSelectChange } from '@angular/material/select';
 import { Environment } from '../../../../common/interface/environment.interface';
+import { isValid } from 'date-fns';
+import { DateFnsAdapter } from '@angular/material-date-fns-adapter';
+import { DatetimeAdapter } from '@mat-datetimepicker/core';
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts:35]: this is incorrect adapter, we should be using angular components picker and `@mat-datetimepicker/core` had to be removed, was that not complete?

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts [hunk 7]
```diff
@@ -286,13 +288,13 @@ export abstract class BaseEditTaskFilterCloudComponent<T> implements OnInit, OnC
 
     onDateChanged(newDateValue: any, dateProperty: TaskFilterProperties) {
         if (newDateValue) {
-            const momentDate = moment(newDateValue, BaseEditTaskFilterCloudComponent.FORMAT_DATE, true);
+            const date = DateFnsUtils.formatDate(new Date(newDateValue), BaseEditTaskFilterCloudComponent.FORMAT_DATE);
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts:291]: makes no sense, you first call new Date, then format to string, then convert to date again

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts [hunk 8]
```diff
@@ -286,13 +288,13 @@ export abstract class BaseEditTaskFilterCloudComponent<T> implements OnInit, OnC
...
             const controller = this.getPropertyController(dateProperty);
-
-            if (momentDate.isValid()) {
-                controller.setValue(momentDate.toISOString(true));
+            if (isValid(new Date(date))) {
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts:293]: building new date again

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts [hunk 9]
```diff
@@ -286,13 +288,13 @@ export abstract class BaseEditTaskFilterCloudComponent<T> implements OnInit, OnC
...
+                controller.setValue(new Date(date).toISOString());
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts:294]: building new date again

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts [hunk 10]
```diff
@@ -137,7 +138,7 @@ export abstract class BaseEditTaskFilterCloudComponent<T> implements OnInit, OnC
 
     constructor(
         protected formBuilder: UntypedFormBuilder,
-        protected dateAdapter: DateAdapter<Moment>,
+        protected dateAdapter: DatetimeAdapter<DateFnsAdapter>,
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/base-edit-task-filter-cloud.component.ts:141]: incorrect type

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/edit-service-task-filter-cloud.component.ts [hunk 11]
```diff
@@ -42,7 +42,7 @@ export class EditServiceTaskFilterCloudComponent extends BaseEditTaskFilterCloud
         dialog: MatDialog,
         translateService: TranslationService,
         private serviceTaskFilterCloudService: ServiceTaskFilterCloudService,
-        dateAdapter: DateAdapter<Moment>,
+        dateAdapter: DatetimeAdapter<DateFnsAdapter>,
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/edit-service-task-filter-cloud.component.ts:45]: incorrect type

File: lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/edit-task-filter-cloud.component.ts [hunk 12]
```diff
@@ -43,10 +43,11 @@ export class EditTaskFilterCloudComponent extends BaseEditTaskFilterCloudCompone
         dialog: MatDialog,
         translateService: TranslationService,
         private taskFilterCloudService: TaskFilterCloudService,
-        dateAdapter: DateAdapter<Moment>,
+        dateAdapter: DatetimeAdapter<DateFnsAdapter>,
```
Comment (DenysVuika) [inline lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filters/edit-task-filter-cloud.component.ts:46]: incorrect type

**Pull request closed without merging**